bitcoin/script/script.h

```diff
@@ -587,4 +573,28 @@ struct CScriptWitness
 
 bool CheckMinimalPush(const std::vector<unsigned char>& data, opcodetype opcode);
 
+/** Build a script by concatenating other scripts, or any argument accepted by CScript::operator<<. */
+template<typename... Ts>
+CScript BuildScript(Ts&&... inputs)
+{
+    CScript ret;
+    size_t cnt{0}; // Try to optimize over repeatedly calling ret.empty()
+
+    ([&] (Ts&& input) {
+        if constexpr (std::is_same_v<std::remove_cv_t<std::remove_reference_t<Ts>>, CScript>) {
+            // If it is a CScript, extend ret with it. If ret is empty, move or copy it instead.
```

Comment:
The comment is out of date now (it's only move/copying the first argument, even if it's still empty after the first one).